pkg/proc/proc.go

```diff
@@ -466,18 +466,22 @@ func StepOut(dbp Process) error {
 
 // GoroutinesInfo returns an array of G structures representing the information
 // Delve cares about from the internal runtime G structure.
-func GoroutinesInfo(dbp Process) ([]*G, error) {
+func GoroutinesInfo(dbp Process, start, count int) ([]*G, int, error) {
 	if _, err := dbp.Valid(); err != nil {
-		return nil, err
+		return nil, 0, err
```

Comment:
I think returning `nextg==0` on failure and when reading is finished is inconvenient. Seeing the loop in command.go it would be more natural if the sentinel value for "you are done" was -1. Then the goroutines command could be a single loop with `for start >= 0`, and assign the `nextg` returned by ListGoroutines to `start`.